nilearn/_utils/segmentation.py

```diff
@@ -355,7 +357,14 @@ def _solve_cg(lap_sparse, B, tol):
     lap_sparse = lap_sparse.tocsc()
     X = []
     for i in range(len(B)):
-        x0 = cg(lap_sparse, -B[i].todense(), tol=tol, atol="legacy")[0]
+        # TODO Python 3.8
+        # consider remove if/else block when dropping support for 3.8
+        # would require pinning scipy to >= 1.12
+        # See https://github.com/nilearn/nilearn/pull/4394
```

Comment:
Will open an issue after merging to keep track of the things to do when we drop python 3.8.